gprofiler/spark/spark_collector.py

```diff
@@ -0,0 +1,894 @@
+        try:
+            # resolve config path against process' filesystem root
+            process_relative_config_path = self._get_process_root_relative_path(
+                process, self._get_yarn_config_path(process)
+            )
+            with open(process_relative_config_path, "rb") as conf_file:
+                config_xml_string = conf_file.read()
+            return ET.fromstring(config_xml_string)
+        except FileNotFoundError:
+            return None
+
+    def _get_yarn_config_property(
+        self, process: psutil.Process, requested_property: str, default: Any = None
+    ) -> Optional[str]:
+        config = self._get_yarn_config(process)
+        if config is not None:
+            for config_property in config.iter("property"):
+                name_property = config_property.find("name")
+                if name_property is not None and name_property.text == requested_property:
+                    value_property = config_property.find("value")
+                    if value_property is not None:
+                        return value_property.text
+        return default
+
+    def _guess_driver_application_master_address(self, process: psutil.Process) -> str:
+        if self._master_address is not None:
+            return self._master_address
+        else:
+            host_name = get_hostname()
+            return host_name + ":4040"
+
+    def _guess_yarn_resource_manager_webapp_address(self, resource_manager_process: psutil.Process) -> str:
+        config = self._get_yarn_config(resource_manager_process)
+
+        if config is not None:
+            for config_property in config.iter("property"):
+                name_property = config_property.find("name")
+                if (
+                    name_property is not None
+                    and name_property.text is not None
+                    and name_property.text.startswith("yarn.resourcemanager.webapp.address")
+                ):
+                    value_property = config_property.find("value")
+                    if value_property is not None and value_property.text is not None:
+                        return value_property.text
+        if self._master_address is not None:
+            return self._master_address
+        else:
+            host_name = self._get_yarn_host_name(resource_manager_process)
+            return host_name + ":8088"
+
+    def _guess_mesos_master_webapp_address(self, process: psutil.Process) -> str:
+        if self._master_address:
+            return self._master_address
+        else:
+            host_name = get_hostname()
+            return host_name + ":5050"
+
+    def _get_yarn_host_name(self, resource_manager_process: psutil.Process) -> str:
+        host_name = self._get_yarn_config_property(resource_manager_process, "yarn.resourcemanager.hostname")
+        return host_name if host_name is not None else get_hostname()
```

Comment:
Does the fallback to `get_hostname` make sense? Is that what would happen when the profiler runs on ResourceManager nodes?

Reply:
I think this function only runs on ResourceManager nodes (and `resource_manager_process` is an RM running on this node)

Reply:
Do you want to remove the parsing of `yarn.resourcemanager.hostname`?

Reply:
Let's keep it - I added the logs to tell us which path actually gets taken. Then in the future we can see what's dead code, and remove. Honestly I don't know which paths are taken now :shrug: